torch/distributions/half_cauchy.py

```diff
@@ -43,7 +43,7 @@ def scale(self):
 
     @property
     def mean(self):
-        return self.base_dist.mean
+        return torch.full(self._extended_shape(), math.inf, dtype=self.scale.dtype, device=self.scale.device)
```

Comment:
@fehiepsi - It turns out that I had not taken a close look and the stats for half-normal/half-cauchy look fine as is. I still made this modification to return `inf` rather than `nan` for the half-cauchy, but let me know if you had any other correction in mind.

Reply:
Thanks, @neerajprad! Looks good to me. You'll need to fix `variance` too.

Reply:
The variance for the base dist should already be `inf` so I left it as is.

Reply:
Oops, you are right!